src/AddressRegistry.Api.Legacy/AddressMatch/Matching/Sanitizer.cs

```diff
@@ -5,376 +5,599 @@ namespace AddressRegistry.Api.Legacy.AddressMatch.Matching
+            // huisnummer als prefix in straatnaam
+            // TODO: [0-9]+ ?
+            if (new Regex("[0-9] ").IsMatch(streetName))
             {
-                var requestStraatnaamWithoutStraatnaam = streetName.Replace(StripStreetName(streetName), "").Replace(",", string.Empty).Replace(" ", string.Empty);
-                //huisnummer als prefix in straatnaam
-                if (new Regex("[0-9] ").IsMatch(streetName))
+                if (possibleIndexInStreet == "0000"
+                    && possibleNumberInStreet != null
+                    && int.TryParse(possibleNumberInStreet, out hnr)
+                    && hnr != 0)
                 {
-                    if (FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam)) == "0000"
-                        && FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)) != null
-                        && Int32.TryParse(FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)), out hnr)
-                        && hnr != 0)
-                    {
-                        houseNumber0 = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
-                    }
-                }
-                //huisnummer met niet-numeriek bisnummer als prefix in straatnaam
-                if (new Regex("[0-9][a-zA-Z] ").IsMatch(streetName) && !new Regex("[0-9]e ").IsMatch(streetName))
-                {
-                    if (FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)) != null
-                        && Int32.TryParse(FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam)), out hnr)
-                        && hnr != 0
-                        && new Regex("[0-9][a-zA-Z]$").IsMatch(requestStraatnaamWithoutStraatnaam))
-                    {
-                        houseNumber0 = FormatRrHouseNumber(LeftIndex(requestStraatnaamWithoutStraatnaam));
-                        rrindex0 = FormatRrIndex(RightIndex(requestStraatnaamWithoutStraatnaam)).ToUpper();
-                    }
-                }
-                //huisnummer als suffix in straatnaam
-                if (new Regex(" [0-9]+$").IsMatch(streetName))
-                {
-                    houseNumber0 = FormatRrHouseNumber(new Regex(" [0-9]+$").Match(streetName).Value);
-                    //TODO Stif: ask business, why only here warning?
-                    Warnings.AddWarning("2", "'Huisnummer' in 'Straatnaam' gevonden.");
+                    formattedParts.HouseNumber = possibleNumberInStreet;
                 }
-                //huisnummer met niet-numeriek bisnummer als suffix in straatnaam
-                if (new Regex(" [0-9]+[a-zA-Z]$").IsMatch(streetName))
+            }
+
+            // huisnummer met niet-numeriek bisnummer als prefix in straatnaam
+            // TODO: [0-9]+ ?
+            if (new Regex("[0-9][a-zA-Z] ").IsMatch(streetName) && !new Regex("[0-9]e ").IsMatch(streetName))
```

Comment:
Should this be `[0-9]+`?

Reply:
In retrospect, this will match `XA ` so `blablaXXA ` is seen as a match too :)

It would still be cleaner with `[0-9]+[a-zA-Z] ` explicitly.